### PR TITLE
Add support for tags, delivery times, recipient variables and test mode

### PR DESCRIPTION
**Pull request by mjrussell**

### CODE REVIEW

File: Mail/Hailgun/Internal/Data.hs [hunk 1]
```diff
@@ -86,23 +94,23 @@ data MessageContent
...
+   , messageCC                 :: [VerifiedEmailAddress]
+   , messageBCC                :: [VerifiedEmailAddress]
+   , messageReplyTo            :: Maybe VerifiedEmailAddress
+   , messageInReplyTo          :: Maybe UnverifiedEmailAddress
+   , messageReferences         :: Maybe B.ByteString
+   , messageAttachments        :: [SpecificAttachment]
+   , messageTags               :: [MessageTag]
+   , messageRecipientVariables :: RecipientVariables
+   , messageDeliveryTime       :: Maybe UTCTime
+   , messageTestMode           :: Bool -- ^ Mailgun will not send the message, however it will still charge the account for it
```
Comment (tippenein) [inline Mail/Hailgun/Internal/Data.hs:110]: `::` aligning is bad for this reason

File: Mail/Hailgun/SendEmail.hs [hunk 2]
```diff
@@ -54,13 +55,36 @@ toSimpleEmailParts message =
...
+         Just t -> [(BC.pack "o:deliverytime", BC.pack $ formatUTC t)]
+
+      testMode = if messageTestMode message
+         then [(BC.pack "o:testmode", BC.pack $ "yes")]
+         else []
+
       replyTo = case messageReplyTo message of
          Nothing -> []
-         Just email -> [(BC.pack "h:reply-to", email)]
+         Just email -> [(BC.pack "h:Reply-to", email)]
```
Comment (tippenein) [inline Mail/Hailgun/SendEmail.hs:87]: Did this matter or ?

**Comment by tippenein** [comment]
Looks good :+1:

File: Mail/Hailgun/SendEmail.hs (hunk 2 above)
Comment (mjrussell) [reply Mail/Hailgun/SendEmail.hs:87]: Im not 100% sure I was working through their docs and noticed they seems to be fairly specific about casing so figured it was best to be consistent

**Pull request merged**